std/file.d

```diff
@@ -450,22 +459,112 @@ version (Windows) private void[] readImpl(scope const(char)[] name, scope const(
             OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL | FILE_FLAG_SEQUENTIAL_SCAN,
             HANDLE.init);
     auto h = trustedCreateFileW(namez, defaults);
-
     cenforce(h != INVALID_HANDLE_VALUE, name, namez);
     scope(exit) cenforce(trustedCloseHandle(h), name, namez);
-    ulong fileSize = void;
-    cenforce(trustedGetFileSize(h, fileSize), name, namez);
-    size_t size = min(upTo, fileSize);
-    auto buf = () @trusted { return GC.malloc(size, GC.BlkAttr.NO_SCAN)[0 .. size]; } ();
+
+              ulong  fileSize     = void;
+    immutable bool   fileSizeOkay = trustedGetFileSize(h, fileSize);
+    immutable size_t maxSize      = fileSizeOkay ? min(fileSize, upTo) : upTo;
+
+    //writef("readFile %s: fileSize %u.\n", name, fileSize);
+
+    // Shamelessly reproducing the rest of the function from Posix version, with adaptations kept to a minimum:
```

Comment:
Perhaps instead of copying the code, the copy loop could be refactored into a platform-agnostic template function, with the primitive read APIs passed as small wrappers.

Reply:
I would love to try that ! Thank you for suggestion !